### ISSUE TITLE
Localizable.strings (UTF-16) not showing a diff

### ISSUE THREAD

**Issue by otri** [ForkIssues/Tracker#808]
When making changes to my Localizable.strings file, it doesn't seem to understand UTF-16 encoding, and show a text diff.  When trying an external Diff tool, it also doesn't correctly launch "Beyond Compare" with the left file.

**Comment by maciejtrybilo** [comment]
+1 Would be very helpful for iOS/MacOS dev.

**Comment by noheger** [comment]
+1 Please add support for diffing UTF-16 .strings files.

**Comment by mcx808** [comment]
+1
I notice that a lot of SQL related tools will output SQL scripts in UTF16 by default, including SSMS & RedGate SQL Compare & SQL Data compare.

**Comment by DanPristupov** [comment]
Git doesn't recognize UTF-16 LE encoding by default. You can fix this by creating a rule (e.g. `*.sql text`) in the `.gitattributes` file. (https://www.git-scm.com/docs/gitattributes)

**Comment by mcx808** [comment]
Should that be enough for Fork to be able to diff UTF16LE files? I made the change, commited the gitattributes but UTF16LE .sql files still show as binaries.

> Git doesn't recognize UTF-16 LE encoding by default. You can fix this by creating a rule (e.g. `*.sql text`) in the `.gitattributes` file. (https://www.git-scm.com/docs/gitattributes)

**Comment by DanPristupov** [comment]
Can you try `*.sql diff` instead of `*.sql text`?

> Should that be enough for Fork to be able to diff UTF16LE files?

That should be enough for **git**. Fork is not really involved in diff producing.

**Comment by mcx808** [comment]
> Can you try `*.sql diff` instead of `*.sql text`?
> 
> > Should that be enough for Fork to be able to diff UTF16LE files?
> 
> That should be enough for **git**. Fork is not really involved in diff producing.

Nope, same for any of the following `*.sql text=auto`, `*.sql text`, `*.sql diff`
However as you say it's down to git, and there's a rabbit hole of stackoverflow posts suggesting it's not so straightforward. I'll have to look into it some other time but I'll post back if I find the solution.

**Comment by robmathers** [comment]
Also running into this, as UTF-16 is the default format for localized `.strings` files, including those created by `genstrings`. As a top-notch Mac dev tool, having out of the box support would definitely be nice.

**Comment by backslash-f** [comment]
This works on my end:

- In your project main directory, create or edit a `.gitattributes` file adding this line:
`
*.strings diff=localizablestrings
`

- In the `.gitconfig` file in your home directory, add these lines:
`
[diff "localizablestrings"]
textconv = "iconv -f utf-16 -t utf-8"
`

Source: https://stackoverflow.com/a/51612850

**Comment by ferben** [comment]
`*.strings diff` works for me, but Fork show also BOM mark in diff: `ÿþ`
Could Fork.app filter the BOM mark (`FE FF` or `FF FE` at first 2 bytes of file)?
